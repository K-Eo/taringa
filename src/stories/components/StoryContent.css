.Story-body img {
  max-width: 100%;
  height: auto;
  display: block;
}

.Story-body iframe {
  max-width: 100%;
}
