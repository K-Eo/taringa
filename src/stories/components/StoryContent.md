Story Content

```js
const data = require('./story.data').story;
<div className="row">
  <div className="col-8">
    <StoryContent {...data} status="success" />
  </div>
</div>
```

Story Content placeholder

```js
<div className="row">
  <div className="col-8">
    <StoryContent status="fetching" />
  </div>
</div>
```
