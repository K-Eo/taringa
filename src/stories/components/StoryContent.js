import React from 'react'
import { humanizeNum } from '../../Utils'

import './StoryContent.css'
import Action from '../../common/Action'

const StoryContent = ({
  control: { status },
  story: { comments, content, downvotes, title, upvotes },
}) => {
  if (status === 'fetching') {
    return (
      <div className="StoryContent-ph card">
        <div className="StoryContent-animated-background">
          <div className="StoryContent-title-gap ph" />
          <div className="StoryContent-subtitle-gap ph" />
          <div className="StoryContent-content-gap ph" />
          <div className="StoryContent-meta-extra ph" />
        </div>
      </div>
    )
  }

  return (
    <div className="card Story-body">
      <div className="card-body">
        {title && <h4 className="card-title ">{title}</h4>}
        {content}
      </div>
      <div className="card-footer bg-light">
        <Action className="mr-4" icon="fa fa-arrow-up">
          {humanizeNum(upvotes)}
        </Action>

        <Action className="mr-4" icon="fa fa-arrow-down">
          {humanizeNum(downvotes)}
        </Action>

        <Action icon="far fa-comment">{humanizeNum(comments)}</Action>
      </div>
    </div>
  )
}

StoryContent.defaultProps = {
  control: { status: 'fetching' },
  story: { comments: [], content: null, downvotes: 0, title: '', upvotes: 0 },
}

export default StoryContent
