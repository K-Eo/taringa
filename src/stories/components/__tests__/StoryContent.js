import React from 'react'

import StoryContent from '../StoryContent'

describe('Story Content', () => {
  it('exists', () => {
    expect(StoryContent).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<StoryContent />)

    expect(wrapper).toMatchSnapshot()
  })
})
