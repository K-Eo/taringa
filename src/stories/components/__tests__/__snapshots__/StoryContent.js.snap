// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Content renders 1`] = `
<div
  className="card Story-body"
>
  <div
    className="card-body"
  >
    <h4
      className="card-title "
    >
      #TITLE
    </h4>
    #CONTENT
  </div>
  <div
    className="card-footer bg-light"
  >
    <Action
      className="mr-4"
      icon="fa fa-arrow-up"
    >
      0
    </Action>
    <Action
      className="mr-4"
      icon="fa fa-arrow-down"
    >
      0
    </Action>
    <Action
      icon="far fa-comment"
    >
      0
    </Action>
  </div>
</div>
`;
